actionmailer/lib/action_mailer/base.rb

```diff
@@ -901,11 +901,8 @@ def self.supports_path? # :doc:
       end
 
       def apply_defaults(headers)
-        default_values = self.class.default.transform_values do |value|
-          compute_default(value)
-        end
-
-        headers_with_defaults = headers.reverse_merge(default_values)
+        headers_with_defaults = headers.reverse_merge(self.class.default)
+        headers_with_defaults = headers_with_defaults.transform_values { |value| compute_default(value) }
```

Comment:
This changes will also call `compute_default` on `headers`, instead of just `self.class.default`.
So this would allow passing procs to mail?
```ruby
mail(from: -> { 'x' })
```

Reply:
Yes you are right the change allows this even though I don't think someone would use it.

If you don't like it maybe we could keep something closer to what was there before by doing something like:
```rb
def apply_defaults(headers)
  default_values = self.class.default.except(*headers.keys).transform_values do |value|
    compute_default(value)
  end

  headers_with_defaults = headers.reverse_merge(default_values)
  headers_with_defaults[:subject] ||= default_i18n_subject
  headers_with_defaults
end
```

Now that I see it, I think it's better if I change it, what do you think?

Reply:
Ah sorry, I finally understand the issue. I shouldn't review on this late at night.
I think the original implementation is fine.